### PR TITLE
[exiv2] ios support

### PR DESCRIPTION
**Pull request by m-kuhn**
**Describe the pull request**

- support for ios
- optional support for nls

### CODE REVIEW

File: ports/exiv2/vcpkg-cmake-wrapper.cmake [hunk 1]
```diff
@@ -2,11 +2,15 @@ _find_package(${ARGS})
 
 if("@VCPKG_LIBRARY_LINKAGE@" STREQUAL "static")
     find_package(Iconv REQUIRED)
-    find_package(Intl REQUIRED)
+    if(@EXIV2_ENABLE_NLS@)
```
Comment (JackBoosY) [inline ports/exiv2/vcpkg-cmake-wrapper.cmake:5]: Please note `EXIV2_ENABLE_NLS` will NOT be defined in `portfile.cmake` using `vcpkg_check_features`, that function will only add `-DEXIV2_ENABLE_NLS=ON|OFF` to `FEATURE_OPTIONS`.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/exiv2/ios.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
+diff --git a/src/futils.cpp b/src/futils.cpp
```
Comment (JackBoosY) [inline ports/exiv2/ios.patch:1]: If it's not the upstream changes, please create a PR in upstream first.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/exiv2/ios.patch (hunk 2 above)
Comment (m-kuhn) [reply ports/exiv2/ios.patch:1]: Done https://github.com/Exiv2/exiv2/pull/1718

Comment (JackBoosY) [reply ports/exiv2/ios.patch:1]: Waiting for upstream approval.

Comment (m-kuhn) [reply ports/exiv2/ios.patch:1]: Upstream merged

File: ports/exiv2/portfile.cmake [hunk 3]
```diff
@@ -7,12 +7,15 @@ vcpkg_from_github(
     REF 194bb65ac568a5435874c9d9d73b1c8a68e4edec #v0.27.3
     SHA512 35a5a41e0a6cfe04d1ed005c8116ad4430516402b925db3d4f719e2385e2cfb09359eb7ab51853bc560138f221900778cd2e2d39f108c513b3e7d22dbb9bf503
     HEAD_REF master
+    PATCHES
+        ios.patch
```
Comment (JackBoosY) [inline ports/exiv2/portfile.cmake:11]:
```suggestion
        ios.patch # Remove this patch in the next update
```

File: ports/exiv2/vcpkg-cmake-wrapper.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/exiv2/vcpkg-cmake-wrapper.cmake:5]: Ping for response.

Comment (m-kuhn) [reply ports/exiv2/vcpkg-cmake-wrapper.cmake:5]: That should do I think?

https://github.com/microsoft/vcpkg/pull/18469/files#diff-1bdf222c873035319124c25b484ba3840ebc8adb43ea4731e43e4627dcf39ac3R57-R61

**Comment by JackBoosY** [comment]
Please apply [this patch](https://dev.azure.com/vcpkg/public/_build/results?buildId=56027&view=artifacts&pathAsName=false&type=publishedArtifacts) to your current branch.

**Comment by JackBoosY** [comment]
When building spix:
```
CMake Error at C:/a/1/s/scripts/buildsystems/vcpkg.cmake:794 (_find_package):
  By not providing "FindQt5.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "Qt5", but
  CMake did not find one.

  Could not find a package configuration file provided by "Qt5" with any of
  the following names:

    Qt5Config.cmake
    qt5-config.cmake

  Add the installation prefix of "Qt5" to CMAKE_PREFIX_PATH or set "Qt5_DIR"
  to a directory containing one of the above files.  If "Qt5" provides a
  separate development package or SDK, be sure it has been installed.
Call Stack (most recent call first):
  lib/CMakeLists.txt:12 (find_package)
```

You should patch _lib/CMakeLists.txt_ to replace `find_package(Qt5)` with the following code that requires your choice:
```
    find_package(Qt5AccessibilitySupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::AccessibilitySupport Qt5::AccessibilitySupport Qt::AccessibilitySupportPrivate Qt5::AccessibilitySupportPrivate)

    find_package(Qt5ClipboardSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::ClipboardSupport Qt5::ClipboardSupport Qt::ClipboardSupportPrivate Qt5::ClipboardSupportPrivate)

    find_package(Qt5Concurrent CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::Concurrent Qt5::Concurrent Qt::ConcurrentPrivate Qt5::ConcurrentPrivate)

    find_package(Qt5Core CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::Core Qt5::Core Qt::CorePrivate Qt5::CorePrivate)

    find_package(Qt5DBus CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::DBus Qt5::DBus Qt::DBusPrivate Qt5::DBusPrivate)

    find_package(Qt5DeviceDiscoverySupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::DeviceDiscoverySupport Qt5::DeviceDiscoverySupport Qt::DeviceDiscoverySupportPrivate Qt5::DeviceDiscoverySupportPrivate)

    find_package(Qt5EdidSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::EdidSupport Qt5::EdidSupport Qt::EdidSupportPrivate Qt5::EdidSupportPrivate)

    find_package(Qt5EventDispatcherSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::EventDispatcherSupport Qt5::EventDispatcherSupport Qt::EventDispatcherSupportPrivate Qt5::EventDispatcherSupportPrivate)

    find_package(Qt5FbSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::FbSupport Qt5::FbSupport Qt::FbSupportPrivate Qt5::FbSupportPrivate)

    find_package(Qt5FontDatabaseSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::FontDatabaseSupport Qt5::FontDatabaseSupport Qt::FontDatabaseSupportPrivate Qt5::FontDatabaseSupportPrivate)

    find_package(Qt5GraphicsSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::GraphicsSupport Qt5::GraphicsSupport Qt::GraphicsSupportPrivate Qt5::GraphicsSupportPrivate)

    find_package(Qt5Gui CONFIG REQUIRED)
    # Note: 8 target(s) were omitted.
    target_link_libraries(main PRIVATE Qt::Gui Qt5::Gui Qt::GuiPrivate Qt5::GuiPrivate)

    find_package(Qt5Network CONFIG REQUIRED)
    # Note: 1 target(s) were omitted.
    target_link_libraries(main PRIVATE Qt::Network Qt5::Network Qt::NetworkPrivate Qt5::NetworkPrivate)

    find_package(Qt5OpenGL CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::OpenGL Qt5::OpenGL Qt::OpenGLPrivate Qt5::OpenGLPrivate)

    find_package(Qt5OpenGLExtensions CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::OpenGLExtensions Qt5::OpenGLExtensions Qt::OpenGLExtensionsPrivate Qt5::OpenGLExtensionsPrivate)

    find_package(Qt5PlatformCompositorSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::PlatformCompositorSupport Qt5::PlatformCompositorSupport Qt::PlatformCompositorSupportPrivate Qt5::PlatformCompositorSupportPrivate)

    find_package(Qt5PrintSupport CONFIG REQUIRED)
    # Note: 1 target(s) were omitted.
    target_link_libraries(main PRIVATE Qt::PrintSupport Qt5::PrintSupport Qt::PrintSupportPrivate Qt5::PrintSupportPrivate)

    find_package(Qt5ServiceSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::ServiceSupport Qt5::ServiceSupport Qt::ServiceSupportPrivate Qt5::ServiceSupportPrivate)

    find_package(Qt5Sql CONFIG REQUIRED)
    # Note: 2 target(s) were omitted.
    target_link_libraries(main PRIVATE Qt::Sql Qt5::Sql Qt::SqlPrivate Qt5::SqlPrivate)

    find_package(Qt5Test CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::Test Qt5::Test Qt::TestPrivate Qt5::TestPrivate)

    find_package(Qt5ThemeSupport CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::ThemeSupport Qt5::ThemeSupport Qt::ThemeSupportPrivate Qt5::ThemeSupportPrivate)

    find_package(Qt5Widgets CONFIG REQUIRED)
    # Note: 1 target(s) were omitted.
    target_link_libraries(main PRIVATE Qt::Widgets Qt5::Widgets Qt::WidgetsPrivate Qt5::WidgetsPrivate)

    find_package(Qt5Xml CONFIG REQUIRED)
    target_link_libraries(main PRIVATE Qt::Xml Qt5::Xml Qt::XmlPrivate Qt5::XmlPrivate)
```

**Comment by m-kuhn** [comment]
Let's wait for upstream to release a new version

**Comment by m-kuhn** [comment]
Minified PR opened https://github.com/microsoft/vcpkg/pull/18964

**Pull request closed without merging**